clevertap-core/src/main/java/com/clevertap/android/sdk/inapp/images/FileResourceProvider.kt

```diff
@@ -0,0 +1,357 @@
+package com.clevertap.android.sdk.inapp.images
+
+import android.content.Context
+import android.graphics.Bitmap
+import com.clevertap.android.sdk.ILogger
+import com.clevertap.android.sdk.inapp.images.memory.FileMemoryAccessObject
+import com.clevertap.android.sdk.inapp.images.memory.InAppGifMemoryAccessObjectV1
+import com.clevertap.android.sdk.inapp.images.memory.InAppImageMemoryAccessObjectV1
+import com.clevertap.android.sdk.inapp.images.memory.MemoryAccessObject
+import com.clevertap.android.sdk.inapp.images.memory.MemoryCreator
+import com.clevertap.android.sdk.inapp.images.memory.MemoryDataTransformationType.MEMORY_DATA_TRANSFORM_TO_BITMAP
+import com.clevertap.android.sdk.inapp.images.memory.MemoryDataTransformationType.MEMORY_DATA_TRANSFORM_TO_BYTEARRAY
+import com.clevertap.android.sdk.inapp.images.memory.MemoryDataTransformationType.MEMORY_DATA_TRANSFORM_TO_FILE
+import com.clevertap.android.sdk.network.DownloadedBitmap
+import com.clevertap.android.sdk.utils.CTCaches
+import java.io.ByteArrayOutputStream
+import java.io.File
+
+internal class FileResourceProvider(
+    private val images: File,
+    private val gifs: File,
+    private val allFileTypesDir: File,
+    private val logger: ILogger? = null,
+    private val inAppRemoteSource: FileFetchApiContract = FileFetchApi()
+) {
```

Comment:
this is a feature specific class and it should only have following methods 

saveFile()
hasFile()/isFIleCached()
getFile(cacheOnly: boolean)

This class can encapsulate another sub feature class which was `InAppResourceProvider`

Can we not have so many methods in this class as it is confusing to understand.

Reply:
resolved